modules/social_features/social_profile/modules/social_profile_privacy/social_profile_privacy.module

```diff
@@ -119,6 +141,9 @@ function social_profile_privacy_admin_settings_form_submit($form, FormStateInter
   }
 
   $config->save();
+
+  // We need to update profile names.
+  SocialProfilePrivacyBatchHelper::bulkUpdateProfileNames();
```

Comment:
Let's add a check for the changes in the form values which are relevant to trigger this update.

Reply:
I have added it.